stock_picking_show_lot/models/stock_picking.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def action_show_lots(self):
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "stock.action_production_lot_form"
+        )
+        action["context"] = {}
+        action["domain"] = [("id", "in", self.move_line_ids.lot_id.ids)]
+        action["name"] = _("View Lots")
+        return action
```

Comment:
Please display a tree view when All lots are more than one, if not, display the form view directly

Reply:
done!